### PR TITLE
Refactors revolution code to be a little less of a mess

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Refactors revolution code to be a little less of a mess.

## Why It's Good For The Game
Revolution code is currently a bit of a mess. Hopefully also fixes issues with deconversion messages not being properly displayed during revolution rounds.

## Changelog
:cl:
No player facing changes
/:cl:

### CODE REVIEW

File: code/game/gamemodes/revolution/revolution.dm [hunk 1]
```diff
@@ -62,10 +60,10 @@
 	var/weighted_score = min(max(round(heads.len - ((8 - sec.len) / 3)),1),max_headrevs)
 
 	while(weighted_score < head_revolutionaries.len) //das vi danya
-		var/datum/mind/trotsky = pick(head_revolutionaries)
-		possible_revolutionaries += trotsky
-		head_revolutionaries -= trotsky
-		update_rev_icons_removed(trotsky)
+		var/datum/mind/removable_headrev = pick(head_revolutionaries)
+		possible_revolutionaries += removable_headrev
+		head_revolutionaries -= removable_headrev
```
Comment (Sirryan2002) [inline code/game/gamemodes/revolution/revolution.dm:65]:
```suggestion
		var/datum/mind/removable_headrev = pick_n_take(head_revolutionaries)
		possible_revolutionaries += removable_headrev
```

File: code/game/gamemodes/revolution/revolution.dm [hunk 2]
```diff
@@ -81,16 +79,14 @@
...
-	if(check_counter >= 5)
-		if(!finished)
-			check_heads()
+//////////////////////////////////////////////////////////////////////////
+//Check if heads and head revolutionaries are still alive and on Z-level//
+//////////////////////////////////////////////////////////////////////////
+/datum/game_mode/revolution/transit_z(mob/living/player)
+	if(!finished)
+		var/list/heads = get_all_heads()
+		if(player.mind && ((player.mind in heads)||(player.mind in head_revolutionaries)))
```
Comment (Sirryan2002) [inline code/game/gamemodes/revolution/revolution.dm:88]:
```suggestion
		if(player.mind && ((player.mind in heads) || (player.mind in head_revolutionaries)))
```

File: code/game/objects/items/devices/flash.dm [hunk 3]
```diff
@@ -155,29 +155,14 @@
...
-
-					if(resisted)
-						to_chat(user, "<span class='warning'>This mind seems resistant to [src]!</span>")
-				else
-					to_chat(user, "<span class='warning'>They must be conscious before you can convert [M.p_them()]!</span>")
-			else
-				to_chat(user, "<span class='warning'>This mind is so vacant that it is not susceptible to influence!</span>")
-
+/obj/item/flash/proc/revolution_conversion(mob/M, mob/user)
+	if(ishuman(M) && (user.mind in SSticker.mode.head_revolutionaries))
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/flash.dm:159]: can you early return this?

Comment (S34NW) [inline code/game/objects/items/devices/flash.dm:159]: Still needs an early return here please

Comment (Adrer) [reply code/game/objects/items/devices/flash.dm]: Added the early return

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/objects/items/devices/flash.dm [hunk 4]
```diff
@@ -155,29 +155,15 @@
...
-						resisted = 1
-
-					if(resisted)
-						to_chat(user, "<span class='warning'>This mind seems resistant to [src]!</span>")
-				else
-					to_chat(user, "<span class='warning'>They must be conscious before you can convert [M.p_them()]!</span>")
-			else
-				to_chat(user, "<span class='warning'>This mind is so vacant that it is not susceptible to influence!</span>")
-
+/obj/item/flash/proc/revolution_conversion(mob/M, mob/user)
```
Comment (farie82) [inline code/game/objects/items/devices/flash.dm:158]: Wait it ain't even a special flash they use? Damn never knew

**Review by farie82 — APPROVED** [review]

**Pull request merged**